src/components/tooltip/tooltip.js

```diff
@@ -187,6 +187,9 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
     }
 
     function setVisible (value) {
+      if ((setVisible.queued && setVisible.visible === !!value) || (!setVisible.queued && scope.visible === !!value)) {
```

Comment:
@kwypchlo I'm trying to understand your logic here.

Won't work the following snippet?
```js
if (setVisible.queued && setVisible.value === value || scope.visible === value) return;
```

It checks if the visibility is in queue and then compares the temporary stored value.
- If `true`: Cancel the logic, because it's the same value and a queue is running -> unnecessary.

If the first condition has passed (no queue / no same value), then we will check the current scope value against the new value.
- If `true`: Cancel the logic, because the tooltip scope has already the assigned value. -> unnecessary step.


But (OR operator) if there is no queue in progress and the tooltip is already set to the same value.